pkg/deploy/vmsscleaner/clean.go

```diff
@@ -0,0 +1,50 @@
+package vmsscleaner
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/compute"
+)
+
+type Interface interface {
+	RemoveFailedNewScaleset(ctx context.Context, rgName, vmssName string) (deleted bool)
+}
+
+type cleaner struct {
+	log  *logrus.Entry
+	vmss compute.VirtualMachineScaleSetsClient
+}
+
+func New(log *logrus.Entry, vmss compute.VirtualMachineScaleSetsClient) Interface {
+	return &cleaner{
+		log:  log,
+		vmss: vmss,
+	}
+}
+
+func (c *cleaner) RemoveFailedNewScaleset(ctx context.Context, rgName, vmssName string) (deleted bool) {
+	scalesets, err := c.vmss.List(ctx, rgName)
+	if err != nil || len(scalesets) < 2 { // Don't remove scaleset if it's the only one
+		return false
+	}
+
+	for _, vmss := range scalesets {
+		if *vmss.Name != vmssName {
+			continue
+		}
```

Comment:
@petrkotas Here is the logic that skips deletion if the VMSS name is not the right one.